### ISSUE TITLE
Multi-chunks on Transform stream

### ISSUE THREAD

**Issue by william-p** [syavorsky/comment-parser#76]
When you are a lot of files to parse and/or big files, stream was truncked. `_transform` was called many times with chunk, and sometime, in middle of block. 

You can see in action with my fork: https://github.com/william-p/comment-parser/blob/bug-bigfiles/tests/bigfiles.spec.js

What do you think ?

**Comment by syavorsky** [comment]
thanks for heads up, will try to investigate and fix it next couple of days

**Comment by syavorsky** [comment]
I couldn't make your tests fail but I see the cause. The `fs.createReadStream` used underneath doesn't guarantee chunks consistency. My first naive attempt was to fix this with having a a custom reader not splitting blocks on middle, but even so each chunk is processed individually and would have line numbers messed up.

I will keep this problem in mind and hopefully come up with better approach later

**Comment by syavorsky** [comment]
so far thinking of consistent reader + initial line number option for parser, that may be the path to take

**Comment by william-p** [comment]
Yes it's tricky ...

Maybe you can put logic in your custom `Transform` (https://github.com/syavorsky/comment-parser/blob/master/index.js#L17) and use `_write` with block detection and `_final` for forward result ?

I'm not `Stream` expert ;)

If you want, i'm available for test somethings in my environment.

**Comment by syavorsky** [comment]
yes, something along those lines. As temporary workaround you can read entire file and pass it to parser instead.

**Comment by william-p** [comment]
Yes, I did this ;)

**Comment by Rybadour** [comment]
Sounds like this might be the root cause of what my team is experiencing. When we call file(fileName, callback) the results from the callback sometimes returns incorrectly parsed tags such as:

```
{
    "tags": [
      {
        "tag": "classprop",
        "type": "string",
        "name": "toolt",
        "optional": false,
        "description": "ip-text This is the text to be shown inside the tooltip placed next to the label text. The tooltip will only appear if this is\nset.",
        "line": 139,
        "source": "@classprop {string} toolt\nip-text This is the text to be shown inside the tooltip placed next to the label text. The tooltip will only appear if this is\nset."
      }
    ],
    "line": 138,
    "description": "",
    "source": "@classprop {string} toolt\nip-text This is the text to be shown inside the tooltip placed next to the label text. The tooltip will only appear if this is\nset."
  },
```

**Comment by syavorsky** [comment]
`1.0` went to the `master` and is current stable version. It is missing the streaming parser, but I am working on it. Parsing the whole source at once should be still the best workaround for now
